Cabal/Distribution/Utils/Glob/Match.hs

```diff
@@ -0,0 +1,106 @@
+module Distribution.Utils.Glob.Match where
+
+import Control.Monad
+    ( (>=>) )
+import Data.Maybe
+    ( listToMaybe )
+import Data.List
+    ( stripPrefix )
+import Distribution.Utils.Glob.Type
+
+isMatch :: Glob -> FilePath -> Bool
+isMatch (Glob realGlob) fp = realIsMatch realGlob fp
+isMatch (NoGlob fp') fp = fp' == fp
+
+realIsMatch :: RealGlob -> FilePath -> Bool
+realIsMatch (RealGlob parts) fp = isMatch' True parts (toSegments fp)
+
+toSegments :: FilePath -> [String]
+toSegments = filter (not . null) . splitOn '/'
+
+-- Not quite the same as the function from Data.List.Split, but this allows
```

Comment:
`Data.List.Split.splitOn` has type `Eq a => [a] -> [a] -> [[a]]` and allows splitting on sublists rather than just individual items. Since we only need the latter here, I thought it would make more sense to just implement this here. I guess I could explain this a bit more in the comments?

Copying the implementation from `Data.List.Split` would involve copying 5 or 6 separate functions, and they all have quite a lot more generality than is needed here - I think it would make this code a bit confusing.